### PR TITLE
Mldb 1409 irls too early stopping

### PR DESCRIPTION
**Pull request by mailletf**

### CODE REVIEW

**Comment by guyd** [comment]
Reviewed but I don't know this code so I don't feel comfortable giving the +1.

File: ml/algebra/least_squares.h [hunk 1]
```diff
@@ -405,7 +405,13 @@ irls(const distribution<Float> & y, const boost::multi_array<Float, 2> & x,
         //                << " tolerance " << tolerence << endl;
         //}
     }
-    
+
+    if(iter >= max_iter) {
```
Comment (jeremybarnes) [inline ml/algebra/least_squares.h]: Are you sure we want to keep this in?

File: ml/jml/testing/weighted_training_test.cc [hunk 2]
```diff
@@ -64,66 +64,77 @@ BOOST_AUTO_TEST_CASE( weighted_training_test )
...
+    vector<string> probabilizer = {"-p 3", "-p 1 -Q 1"};
+    for(int probIdx=0; probIdx<probabilizer.size(); probIdx++) {
+        cout << ML::format(" ----------------------- TESTING WITH PROBABILIZER IN MODE %s (%s) "
+                "-----------------------", probabilizer[probIdx], (probIdx == 0 ? "off" : "on")) << endl;
 
-    string cmd2 = "build/x86_64/bin/classifier_training_tool " + filename2 + " -c " + configname + " -n glz glz.link_function=linear glz.ridge_regression=false -E 1 -p 1 -o " + modelname2 + " glz.verbosity=10  -W LABEL/F,ExampleId/F,WEIGHT/V -z ExampleId -z WEIGHT";
+        // weights: LABEL, ExampleId
+        // ignore: ExampleId
+        string cmd1 = "make classifier_training_tool; build/x86_64/bin/classifier_training_tool " + filename1 + " -c " + configname + " -n glz glz.link_function=linear glz.ridge_regression=false -E 1 "+probabilizer[probIdx]+" -o " + modelname1 + " glz.verbosity=10  -W LABEL/F,ExampleId/F -z ExampleId";
```
Comment (jeremybarnes) [inline ml/jml/testing/weighted_training_test.cc]: We should remove the "make" step, and instead make $(TESTS)/weighted_training_test depend on $(BIN)/classifier_training_tool in the makefile.

File: ml/algebra/least_squares.h (hunk 1 above)
Comment (jeremybarnes) [inline ml/algebra/least_squares.h]: (I mean the debugging)

**Comment by jeremybarnes** [comment]
+1 from me.  Let's finish our discussion on the expected weighted probabilizer behaviour.

**Pull request merged**